src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -249,6 +250,32 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
         } else if (std::dynamic_pointer_cast<ConvertLike>(op)) {
             is_changed |= inherit_output_shape(op, {0});
             is_changed |= inherit_output_type(op, {1});
+        } else if (std::dynamic_pointer_cast<Transpose>(op)) {
+            auto transpose_order = get_constant_from_source(op->input_value(1));
+            if (output_shape.rank().is_static()) {
+                if (transpose_order) {
+                    // set more precise dimensions during reverse infer
+                    // if transpose order is known
+                    int64_t rank_length = output_shape.rank().get_length();
```

Comment:
Could you also add test for this case when order is known?

Reply:
added `TransposeWithConstantOrderReverseInfer`

Reply:
Added